packages/shell-api/src/result.ts

```diff
@@ -108,28 +107,14 @@ export class DeleteResult extends ShellApiClass {
   }
 }
 
-class PrintableCursorIterationResult {
-  documents: Document[];
-  cursorHasMore: boolean;
-  constructor(cursor: CursorIterationResult) {
-    this.documents = [...cursor];
-    this.cursorHasMore = cursor.hasMore;
-  }
-}
-
-// NOTE: because this is inherited, the decorator does not add attributes. So no help() function.
 @shellApiClassDefault
-export class CursorIterationResult extends Array {
-  [shellApiType]: string;
-  hasMore: boolean;
+export class CursorIterationResult extends ShellApiClass {
```

Comment:
Do we still need this to be a class? perhaps we can just ditch this thing / turn it into an interface now that there is no special behaviour, would be one concept less to understand.

Reply:
I don’t think turning this into an interface would make this easier to deal with necessarily – for now, we’re at least still using the feature that this automatically assigns the right shell API type this way.

(Or, another way around … I think if we got rid of this, I think we’d still want to have a factory function that generates these objects in order to avoid code duplication, and classes are basically factory functions)